### DISCUSSION TITLE
Section 8 : Transfer Learning Setup Confusion Matrix

### DISCUSSION THREAD

**Discussion by pusapatiakhilraju** [mrdbourke/pytorch-deep-learning#487] [Q&A] [answered]
Hello Everyone,

I am trying to implement the confusion matrix after implementing transfer learning to see how the model is performing. I am unable to do it. Can someone help?
<img width="1035" alt="1" src="https://github.com/mrdbourke/pytorch-deep-learning/assets/26441101/826410e5-5fc8-4036-a74a-0c87ba3ee141">
<img width="1030" alt="2" src="https://github.com/mrdbourke/pytorch-deep-learning/assets/26441101/da538716-9e9d-4cc7-a0be-bd9bf9927e66">
In screenshot 2 I get the error ''list' object has no attribute 'ndim'' at
this part of the code -> confmat_tensor = confmat(preds=y_pred_tensor,
                         target=test_data.targets)

**Comment by pusapatiakhilraju** [comment, marked as answer]
Okay, I fixed it. 
If anyone is interested on how I've done it then the solution is as follows: Confusion Matrix expects both y_preds and test_data.targets to be tensors with dimensions.
In my scenario you can see in confmat_tensor = confmat(preds=y_pred_tensor, target=test_data.targets) that the test_data.targets was a list.

Hence, I have added the following lines of code to change them into tensors with dimensions
# Convert list of tensors to a single tensor because confusion matrix expects preds and targets to be tensors with dimensions.
y_pred_tensor = torch.cat(y_preds)
y_true_tensor = torch.tensor(test_data.targets)
And now it works
<img width="1050" alt="Screenshot 2023-06-15 at 10 48 24 AM" src="https://github.com/mrdbourke/pytorch-deep-learning/assets/26441101/56e85745-6330-478c-b7e4-f3da3ebd4a6c">

**Reply by mrdbourke** [reply]
Glad to see you've fixed it @pusapatiakhilraju ! Thank you for the update :)
